src/hooks/useActiveLocale.ts

```diff
@@ -14,6 +16,25 @@ function parseLocale(maybeSupportedLocale: string): SupportedLocale | undefined
   )
 }
 
+/**
+ * Returns the supported locale read from the url query parameter (lng=)
+ */
+function urlLocale(): SupportedLocale | undefined {
+  const search =
+    location.search ||
```

Comment:
this feel hacky. Can you not use react-router-dom?

Reply:
This is called outside of React, so I don't think any react-router-dom's are yet available.

Reply:
nvm, just found https://reactrouter.com/web/api/matchPath, will update.

Reply:
I misread - matchPath will not suffice here.

I don't see how to use react-router-dom here, as it is outside of the React render cycle. Is there an underlying parser that react-router-dom uses that we can also rely on, to avoid this?

Reply:
We only use the HashRouter, so I've updated this to only check the hash, and parse the query from there. That's fairly straightforward, and similar to what we do in other query-parsing scenarios.

Reply:
I just don't really like the precedent this sets. I don't think we should be hacking the tools we use to save 10ms. e.g. what if we move away from react-router? This code would start failing and we have no test coverage either.

Could we consider wrapping this code with `withRouter` to be able to grab the location from the react-router `location` object?